### PR TITLE
chore: Add a section on how to verify Bitcoin canister upgrade proposals

### PR DESCRIPTION
**Pull request by dsarlis**
Thank you for your contribution to the IC Developer Portal. This repo contains the content for https://internetcomputer.org and the ICP Developer Documentation, https://internetcomputer.org/docs/. 

If you are submitting a Pull Request for adding or changing content on the ICP Developer Documentation, please make sure that your contribution meets the following requirements:

- [x] Follows the [developer docs style guide](style-guide.md).
- [x] Follows the [best practices and guidelines](README.md#best-practices).
- [x] New documentation pages include [document tags](README.md#document-tags).
- [x] New documentation pages include [SEO keywords](README#seo-keywords).
- [x] New documents are in the `.mdx` file format to support the previous two components. 
- [x] New documents are registered in [`/sidebars.js`](https://github.com/dfinity/portal/blob/master/sidebars.js), otherwise, it will not appear in the
  side navigation bar.
- [x] Make sure that the [`.github/CODEOWNERS`](https://github.com/dfinity/portal/blob/master/.github/CODEOWNERS) file is
  filled with new documents that you added. This way we can ensure that future Pull Requests are reviewed by the right people.

### CODE REVIEW

File: docs/developer-docs/daos/nns/proposal-advanced.mdx [hunk 1]
```diff
@@ -102,6 +102,16 @@ The SNS also uses the ICRC ledger suite, which you can find in [the Rosetta repo
...
+
+* You can find the code for the Bitcoin Canister in [the bitcoin-canister repository](https://github.com/dfinity/bitcoin-canister).
+* These proposals are not expected to happen frequently, but rather when there are important updates to be applied to the Bitcoin canister. You can expect that a new release will be cut in preparation of an upgrade in the above repository. See this [example release](https://github.com/dfinity/bitcoin-canister/releases/tag/release%2F2024-04-26).
+* There are 2 Bitcoin canisters, one following the Bitcoin testnet and one following the Bitcoin mainnet network. The proposals will be specific to one of these canisters. Typically, you can expect that the proposal to upgrade the Bitcoin testnet happens 1 week before the one to upgrade Bitcoin mainnet.
+* You can find the instructions on how to verify that the Wasm in the proposal was indeed build from the provided `git commit` in the proposal's section "How can I verify the Wasm hash of this canister?" (see [this example proposal](https://dashboard.internetcomputer.org/proposal/129499)).
```
Comment (ielashi) [inline docs/developer-docs/daos/nns/proposal-advanced.mdx:112]:
```suggestion
* You can find the instructions on how to verify that the Wasm in the proposal was indeed built from the provided `git commit` in the proposal's section "How can I verify the Wasm hash of this canister?" (see [this example proposal](https://dashboard.internetcomputer.org/proposal/129499)).
```

**Review by ielashi — APPROVED** [review]

**Review by LaraAS — APPROVED** [review]
That is great, thanks for the quick contribution!

File: docs/developer-docs/daos/nns/proposal-advanced.mdx [hunk 2]
```diff
@@ -102,6 +102,16 @@ The SNS also uses the ICRC ledger suite, which you can find in [the Rosetta repo
 * SNS upgrade proposals are announced in [this forum thread](https://forum.dfinity.org/t/sns-upgrade-aggregation-thread/24259).
 * You can find the instructions on how to verify that the Wasm in the proposal was indeed build from the provided `git commit` in the proposal's section "Wasm Verification" (see [this example proposal](https://dashboard.internetcomputer.org/proposal/129633)).
 
+**How to verify Bitcoin canister upgrades**
+
+If you are looking to verify Bitcoin canister upgrades apart from regularly checking open proposals, you might want to consider the following information:
+
+* You can find the code for the Bitcoin Canister in [the bitcoin-canister repository](https://github.com/dfinity/bitcoin-canister).
```
Comment (jessiemongeon1) [inline docs/developer-docs/daos/nns/proposal-advanced.mdx:109]:
```suggestion
* You can find the code for the Bitcoin canister in [the `bitcoin-canister` repository](https://github.com/dfinity/bitcoin-canister).
```

File: docs/developer-docs/daos/nns/proposal-advanced.mdx [hunk 3]
```diff
@@ -102,6 +102,16 @@ The SNS also uses the ICRC ledger suite, which you can find in [the Rosetta repo
...
+* These proposals are not expected to happen frequently, but rather when there are important updates to be applied to the Bitcoin canister. You can expect that a new release will be cut in preparation of an upgrade in the above repository. See this [example release](https://github.com/dfinity/bitcoin-canister/releases/tag/release%2F2024-04-26).
+* There are 2 Bitcoin canisters, one following the Bitcoin testnet and one following the Bitcoin mainnet network. The proposals will be specific to one of these canisters. Typically, you can expect that the proposal to upgrade the Bitcoin testnet happens 1 week before the one to upgrade Bitcoin mainnet.
```
Comment (jessiemongeon1) [inline docs/developer-docs/daos/nns/proposal-advanced.mdx:111]:
```suggestion
* There are 2 Bitcoin canisters, one following the Bitcoin testnet and one following the Bitcoin mainnet network. The proposals will be specific to one of these canisters. Typically, you can expect that the proposal to upgrade the Bitcoin testnet canister happens 1 week before the one to upgrade the Bitcoin mainnet canister.
```

File: docs/developer-docs/daos/nns/proposal-advanced.mdx [hunk 4]
```diff
@@ -102,6 +102,16 @@ The SNS also uses the ICRC ledger suite, which you can find in [the Rosetta repo
...
+* You can find the instructions on how to verify that the Wasm in the proposal was indeed built from the provided `git commit` in the proposal's section "How can I verify the Wasm hash of this canister?" (see [this example proposal](https://dashboard.internetcomputer.org/proposal/129499)).
+
+For more information about the Bitcoin canister, you can read the [Bitcoin Canister documentation](https://github.com/dfinity/bitcoin-canister/blob/master/README.md).
```
Comment (jessiemongeon1) [inline docs/developer-docs/daos/nns/proposal-advanced.mdx:114]:
```suggestion
For more information about the Bitcoin canister, you can read the [Bitcoin canister documentation](https://github.com/dfinity/bitcoin-canister/blob/master/README.md).
```

**Review by jessiemongeon1 — APPROVED** [review]

**Pull request merged**